### PR TITLE
#105 Update all image paths linked to /wp-content/*

### LINKED ISSUE #105: Update all old image links if necessary
### Problem

It's not clear whether this is a problem actually, however there might be.

After #76 is closed, a new folder will exist in this repo: `public/wp-content/uploads`

All of the blog post images currently load images from the old wordpress site.

Since I copied all of those images and put them in the `public/` folder, theoretically they should just load when the new site is deployed.

However one thing that will definitely break is the use of `/groupincome.org/` on the test GH deployment page. Perhaps that then should be fixed?

### Solution

Investigate how to get the images loading properly from the local repo, including the new test/GH deployment site.

### PR DESCRIPTION
**Pull request by SebinSong**
closes #105 

To use `resolvePath()` utility function within various `.md` files, they had to be first turned into `.mdx` (MD + JSX) files, which is what large portion of this PR has.

and some of the `.md` specific syntax had to be turned into the `jsx` equivalent along with that.

e.g.)
```
![alt name](url)
```
->
```
<img src={resolvePath('url')} alt='alt name' />
```
Thanks,

### CODE REVIEW

File: astro.config.mjs [hunk 1]
```diff
@@ -3,6 +3,8 @@ import mdx from '@astrojs/mdx';
 import sitemap from '@astrojs/sitemap';
 import remarkBreaks from 'remark-breaks'; // improves support for newlines in markdown files
 import remarkGfm from 'remark-gfm'; // support rendering tables in markdown files
+import remarkMath from 'remark-math';
+import rehypeKatex from 'rehype-katex';
```
Comment (SebinSong) [inline astro.config.mjs:7]: This is a fix for a warning that occurs in the command line tools as soon as `mdx()` plugin below is enabled.

File: src/pages/[...blog].astro [hunk 2]
```diff
@@ -19,59 +19,65 @@ const { Content } = Astro.props
 <Content />
 
 <script>
-const footnoteEl = document.querySelector('section.footnotes')
+window.addEventListener('DOMContentLoaded', () => {
+  setTimeout(prepFootnotes, 100)
+})
```
Comment (SebinSong) [inline src/pages/[...blog].astro:27]: The footnote related fix merged in PR #90 somehow breaks in any blog posts that are turned into `.mdx` files. But giving the logic a reasonable amount of delay makes it work in both `.md` and `.mdx` files. (Let me know if this needs to be commented within the file itself)

File: src/posts/misconceptions-about-majority-rule.mdx [hunk 3]
```diff
@@ -15,38 +15,41 @@ categories:
...
 
-<!--
+{/*
 FYI, this is another example of academia gone wrong:
 http://www-groups.dcs.st-and.ac.uk/history/PrintHT/Voting.html
 
 - [ ] Post followup two days later.
 - [ ] post job listing to: https://govuejs.com/jobs/1/full-stack-engineer
--->
+*/}
```
Comment (SebinSong) [inline src/posts/misconceptions-about-majority-rule.mdx:26]: `<!-- ... -->` is an illegal syntax in `MDX` ([reference](https://mdxjs.com/docs/what-is-mdx/#mdx-syntax)) so turned it into a JSX equivalent, which is `/* ... */`

File: src/pages/[...blog].astro (hunk 2 above)
Comment (taoeffect) [reply src/pages/[...blog].astro:27]: Yeah, all unusual or odd or weird behavior needs comments 👍

**Review by taoeffect — APPROVED** [review]
LGTM!

**Pull request merged**